indico/core/signals/lowlevel.py

```diff
@@ -0,0 +1,17 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2021 CERN
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see the
+# LICENSE file for more details.
+
+from blinker import Namespace
+
+
+_signals = Namespace()
+
+db_query = _signals.signal('db-query', '''
+The *sender* is a string identifying the context.  The *query* kwarg contains
+the query object to be customized.  The *context* kwarg contains a dictionary
+with additional values that may be required to customize the query.
```

Comment:
Not sure if we need the context kwarg - I'd just pass the context directly as signal kwargs...